server/api/migrations/versions/38b2f0b84eef_add_alert_history.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""add alert history
+
+Revision ID: 38b2f0b84eef
+Revises: fcf2ea01f99a
+Create Date: 2024-10-29 18:10:32.458406
+
+"""
+
+import os
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import mysql
+
+# revision identifiers, used by Alembic.
+revision = "38b2f0b84eef"
+down_revision = "fcf2ea01f99a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "alert_history",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column(
+            "activation_time", mysql.DATETIME(timezone=True, fsp=3), nullable=False
+        ),
+        sa.Column(
+            "name", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "project", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column("data", sa.JSON(), nullable=True),
+        sa.Column(
+            "entity_id", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "entity_kind",
+            sa.String(length=255, collation="utf8mb3_bin"),
+            nullable=False,
+        ),
+        sa.Column(
+            "event_kind", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "severity", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column("number_of_events", sa.Integer(), nullable=False),
+        sa.PrimaryKeyConstraint("activation_time", "id", name="_alert_history_uc"),
+    )
+    # ### end Alembic commands ###
+
+    # Add initial partitioning
+    partition_interval = os.getenv("PARTITION_INTERVAL", "YEARWEEK")
+    default_partition_value = os.getenv("DEFAULT_PARTITION_VALUE", "2023-01-01")
+    partition_sql = f"""
+        ALTER TABLE alert_history
+        PARTITION BY RANGE ({partition_interval}(activation_time)) (
+            PARTITION p0 VALUES LESS THAN ({partition_interval}('{default_partition_value}'))
+        );
+    """
```

Comment:
was it not possible to achieve the partitioning using the orm definitions?

Reply:
@liranbg no, it's not supported in orm